pkgs/ipxe/default.nix

```diff
@@ -1,4 +1,15 @@
-{ stdenv, writeText, perl, xz, openssl, src, lib }:
+{
+  efi ? true,
+  legacy ? true,
+  lib,
+  openssl,
+  perl,
+  src,
+  stdenv,
+  writeText,
+  xz
+}:
```

Comment:
Considering that this is a file that's intended to be called through `pkgs.callPackage`, does it make sense to include non-dependency parameters here? 

First, what's the intended way for a user to change the default values? Call this file with `pkgs.callPackage ./default.nix { legacy = false; }`? Is that how it's usually done? (ping @tfc)

Second, what happens if `pkgs` for some reason ever has a non-boolean attribute `legacy` or `efi`? Wouldn't this then be called with that non-boolean argument, in turn breaking this package?  Or even worse, what if `pkgs` ever has a boolean attribute, which overwrites the default arguments, subtly breaking how this package is being built?

Reply:
Yes, that is how it's usually done. It would be nicer if there was a different convention, because this comes with the problems that you mention.

Reply:
ok, if that's the convention, this should be fine